control-plane/pkg/security/config.go

```diff
@@ -112,5 +116,10 @@ func (cmp *MTConfigMapSecretLocator) SecretName() (string, bool) {
 }
 
 func (cmp *MTConfigMapSecretLocator) SecretNamespace() (string, bool) {
+	if cmp.UseNamespaceInConfigmap {
+		v, ok := cmp.Data[AuthSecretNamespaceKey]
+		return v, ok
+	}
```

Comment:
> if no AuthSecretNamespaceKey is defined in the configmap, we return the configmap namespace.

If your proposal above is what we're doing, shouldn't the condition be the one below?

```suggestion
	if cmp.UseNamespaceInConfigmap {
		if v, ok := cmp.Data[AuthSecretNamespaceKey]; ok && len(v) > 0 {
			return v, ok
		}
	}
```

qq, what's the current KafkaChannel behavior?

I like the "fallback to CM namespace" approach since in Knative we always assume that a reference without a namespace is referring to the object namespace, so that's added consistency.

Notes:
The `ok` we return here decides if we have auth enabled or not, so I think we should assume that if there is the name (returned by `SecretName`), we can figure out the namespace by either reading the CM or using the CM's namespace.

Reply:
> qq, what's the current KafkaChannel behavior?

I found my answer :smile: 

> Default behaviour is as before: secret namespace is the configmap's namespace